rasa/core/policies/ted_policy.py

```diff
@@ -590,6 +633,56 @@ def train(
             shuffle=False,  # we use custom shuffle inside data generator
         )
 
+    def run_post_training_procedures(
+        self, model_data: RasaModelData, label_ids: np.ndarray
+    ) -> None:
+        """Run any tasks to be completed after the model has been trained.
+
+        Args:
+            model_data: Featurized data on which the model has been trained.
+            label_ids: Label ids corresponding to the featurized data.
+        """
+        # No post training procedure for `TEDPolicy`
+        return
```

Comment:
But why do we define this method and even call it from `train` method?
`run_post_training_procedures` is not a part of `Policy` class specification (thus it doesn't need to be defined) so it seems like it can just be removed from here.

Reply:
`IntentTEDPolicy` is derived from `TEDPolicy` and re-uses the `train` method, but only overwrites this `run_post_training_procedures`. Makes sense to me, but it could also be done by doing a `super` call followed by `run_post_training_procedures` inside `IntentTEDPolicy`.

Reply:
I can make a `super()` call as well if that's preferred by both of you.

Reply:
I think `super` would be nicer.

Reply:
The problem with call to `super` is that `train` method of `IntentTEDPolicy` won't have access to `model_data` and `label_ids` which are needed for running the post training procedures. It will have to construct the two on its own resulting in duplicate code.

Reply:
Or else, I can make it part of the `run_training` method. I think this would be better. Will do it now.